main/main.cpp

```diff
@@ -1182,6 +1184,37 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 			OS::get_singleton()->disable_crash_handler();
 		} else if (I->get() == "--skip-breakpoints") {
 			skip_breakpoints = true;
+		} else if (I->get() == "-xr") {
+			// short form switch just turns XR on
+#ifdef TOOLS_ENABLED
+			XRServer::set_xr_mode(XRServer::XRMODE_MIXED);
+#else
+			XRServer::set_xr_mode(XRServer::XRMODE_ON);
+#endif
+		} else if (I->get() == "--xr-mode") {
+			// long form switch allows us to set a specific mode
+			if (I->next()) {
+				String xr_mode = I->next()->get().to_lower();
+				N = I->next()->next();
+				if (xr_mode == "default") {
+					XRServer::set_xr_mode(XRServer::XRMODE_DEFAULT);
+				} else if (xr_mode == "off") {
+					XRServer::set_xr_mode(XRServer::XRMODE_OFF);
+				} else if (xr_mode == "on") {
+					XRServer::set_xr_mode(XRServer::XRMODE_ON);
+#ifdef TOOLS_ENABLED
+				} else if (xr_mode == "mixed") {
+					XRServer::set_xr_mode(XRServer::XRMODE_MIXED);
+#endif
```

Comment:
This should match the logic above for `-xr`.

Reply:
Yeah I think I'm going to change this, indeed makes more sense, I was over complicating this.